core/planner.c

```diff
@@ -1099,14 +1101,12 @@ bool plan(struct deco_state *ds, struct diveplan *diveplan, struct dive *dive, i
 	}
 
 	if (prefs.surface_segment != 0) {
-		for (int i = 0; i < MAX_CYLINDERS; i++)
-			if (cylinder_nodata(&dive->cylinder[i])) {
-				// Switch to an empty air cylinder for breathing air at the surface
-				// If no empty cylinder is found, keep using last deco gas
-				current_cylinder = i;
-				dive->cylinder[i].cylinder_use = NOT_USED;
-				break;
-			}
+		// Switch to an empty air cylinder for breathing air at the surface
+		// If no empty cylinder is found, keep using last deco gas
+		cylinder_t cyl = { 0 };
+		cyl.cylinder_use = NOT_USED;
+		add_to_cylinder_table(&dive->cylinders, dive->cylinders.nr, cyl);
```

Comment:
maybe you can make this return a pointer to the newly added cylinder (so you can combine this with the next line)?

Reply:
That wouldn't make much sense here as `current_cylinder` is an integer not a pointer.

Anyhow, the whole table-functionality should be reworked and returning an iterator to the `add_to_*_table()` functions is a good idea. Let's keep that for a future PR.